.github/workflows/test.yml

```diff
@@ -3,14 +3,32 @@
 
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+      - dev
+
+  pull_request:
+    branches:
+      - dev
+      - main
 
 env:
   TEST_PYTHON_VERSION: 3.9
   TEST_OS: 'ubuntu-20.04'
 
 jobs:
+  check-draft:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Exit if draft PR
+        if: ${{ github.event.pull_request.draft == true }}
+        run: |
+          echo "This is a draft pull request. Exiting."
+          exit 1
   build:
+    needs: check-draft
```

Comment:
Can you describe it? Does this works in all cases?
If it has corner cases we can ignore it. It's not necessary.

Reply:
@sadrasabouri 

Yes, this approach works in all cases. The key difference with filtering in the `on` section is that, technically, the action still triggers but halts after internal checks, preventing further execution. By applying filtering directly in the `on` command, the action is never triggered in the first place.

Additionally, the code defines a pre-job that runs before other jobs. This pre-job checks whether the pull request is a draft, and if it is, the job exits with a non-zero status, preventing subsequent jobs from running.